src/mbgl/layout/symbol_layout.cpp

```diff
@@ -738,10 +738,12 @@ std::vector<float> SymbolLayout::calculateTileDistances(const GeometryCoordinate
                 sumForwardLength += util::dist<float>(line[i + 1], line[i]);
             }
         }
-        for (std::size_t i = ++segment; i-- > 0;) {
+        for (std::size_t i = segment;; --i) {
             tileDistances[i] = sumBackwardLength;
             if (i != 0u) {
                 sumBackwardLength += util::dist<float>(line[i - 1], line[i]);
+            } else {
+                break;
```

Comment:
`break` shall not be needed^

Reply:
If we use std::size_t instead of int, the `break` is necessary, since when i == 0, no matter of `--i` or `i--`, it will invoke ` 0 - 1`, it causes unsigned integer overflow. I will change to `int` to avoid confusions.